app/controllers/api/orchestration_templates_controller.rb

```diff
@@ -6,5 +6,15 @@ def delete_resource(type, id, data = {})
       super
       resource.raw_destroy if resource.kind_of?(OrchestrationTemplateVnfd)
     end
+
+    def copy_resource(type, id, data = {})
+      klass    = collection_class(type)
+      resource = resource_search(id, type, klass)
+      new_resource = resource.dup
```

Comment:
rather than using `.tap(&:save!)` below, you can just tap the whole thing.
```
resource.dup.tap do |new_resource|
   new_resource.assign_attributes(data)
   new_resource.save!
end 
```